File: Project/Web-SPA/src/app/entities/airline.ts
import { Flight } from './flight';

export class Airline {

    id: number;
    adminid: number;
    name: string;
    address: string;
    promoDescription: Array<string>;
    flightDestionations: Array<string>;
    flights: Array<Flight>;
    averageRating: number;
    rates: Array<number>;

    constructor(name: string, address: string) {
        this.name = name;
        this.address = address;
        this.promoDescription = new Array<string>();
        this.flightDestionations = new Array<string>();
        this.flights = new Array<Flight>();
        this.averageRating = 0.0;
        this.rates = new Array<number>();
    }

    rateAriline(rate: number) {
        this.rates.push(rate);
        let sum = 0.00;

        this.rates.forEach(element => {
            sum += element;
        });

        this.averageRating = sum / this.rates.length;
    }
}

File: Project/WebApi/WebApi/Models/Airline.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Runtime.InteropServices;
using System.Threading.Tasks;

namespace WebApi.Models
{
    public class Airline
    {

        public Airline()
        {
            Destinations = new HashSet<AirlineDestionation>();
            Flights = new HashSet<Flight>();
            Rates = new HashSet<AirlineRate>();
            SpecialOffers = new HashSet<SpecialOffer>();
        }
        public int AirlineId { get; set; }
        public string AdminId { get; set; }
        public AirlineAdmin Admin { get; set; }

        public string Name { get; set; }
        public string PromoDescription { get; set; }
        public byte[] LogoUrl { get; set; }
        public virtual Address Address { get; set; }

        public virtual ICollection<AirlineDestionation> Destinations { get; set; }

        public virtual ICollection<Flight> Flights { get; set; }

        public virtual ICollection<AirlineRate> Rates { get; set; }

        public ICollection<SpecialOffer> SpecialOffers { get; set; }
    }
}
